### ISSUE TITLE
Dependencies don't install in Ubuntu

### ISSUE THREAD

**Issue by CaseyHaralson** [r-pufky/ansible_pihole#17]
When trying to install the dependencies task, the "ansible_distribution" conditional reports "Ubuntu" when running against Ubuntu.

**Comment by r-pufky** [comment]
Currently on vacation will take a look. This was explicitly set to fail for ubuntu installations as I do not run ubuntu anymore and can't guarantee compatibility.

Seeing as how this is now being installed on ubuntu I'm for getting support added in, but would need someone with ubuntu experience to confirm /work through any ubuntu specific issues, much like redhat support.

Do you think you can handle ubuntu cases if they come up?

**Comment by CaseyHaralson** [comment]
I'm currently just testing Ansible before pointing it towards any real devices. I have an ubuntu server installed on a VM and was testing pihole there. I actually got everything installed and mirroring the install on my pi, so I'm going to point ansible at my pi soon.

The only change I had to make to this project is what I put in a pull request, and that got everything working from this project. I did have to make some ubuntu server changes for the actual pihole software to install smoothly: turn off ipv6 and something else...some setting or package was missing. I can't remember now.

Now that I have it working, I doubt I will use ubuntu again for this. Do you think you would need more testing/work than what I've done currently to get it working? Sorry, I'm just getting started with linux beyond ubuntu desktop and Docker containers so I'm not sure what sort of issues could appear between different distributions.

**Comment by r-pufky** [comment]
No worries. Thanks for the info.

Those specific changes for ubuntu to get pihole installed are exactly why the current role hard fails outside of pure debian releases (hard fail safe). All those changes that were made to get the role working would need to be implemented in the role by someone to support it.

Pretty simple if it is just packages or specific options, but I don't know what those are, and there is some component of future time figuring out any changes in the role for future ubuntu/pihole releases - a maintainer would be expected to be given the "figure out what needs to be done and do it" problem for the OS, which may be a little too much if you're just getting into the space.

I'm still open to Ubuntu support or other OS support if there is someone to put in the work for that OS.

That being said, it seems like this should be closed and not merged.

**Comment by CaseyHaralson** [comment]
Well, this role doesn't fail on purpose. It fails when the pihole installer borks because of ipv6. The pihole installer fails with a "Ubuntu 22.04 isn't supported" message, but it actually is in their supported list. I found a few threads on the topic and it seems like that is an issue someone was working on.

The only problem I found with this role is that it doesn't install the needed dependencies. If you think this role should fail on purpose, I would suggest adding an error condition before the dependency check that halts the role.
